core/commands/get.go

```diff
@@ -166,12 +166,20 @@ func getCompressOptions(req cmds.Request) (int, error) {
 	return gzip.NoCompression, nil
 }
 
-func get(ctx context.Context, node *core.IpfsNode, p string, compression int) (io.Reader, error) {
+func get(ctx context.Context, node *core.IpfsNode, p string, compression int) (io.Reader, uint64, error) {
 	pathToResolve := path.Path(p)
 	dagnode, err := core.Resolve(ctx, node, pathToResolve)
 	if err != nil {
-		return nil, err
+		return nil, 0, err
 	}
 
-	return utar.NewReader(pathToResolve, node.DAG, dagnode, compression)
+	reader, err := utar.NewReader(pathToResolve, node.DAG, dagnode, compression)
+	ns, err := dagnode.Stat()
+	if err != nil {
+		return nil, 0, err
+	}
+
+	length := uint64(ns.CumulativeSize)
```

Comment:
afaik add progressbar doesn't use ipfs size, so it's fine. I don't know which one dagreader uses.